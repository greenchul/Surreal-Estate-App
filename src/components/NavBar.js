import React from "react";
import "../styles/NavBar.css";

const NavBar = () => {
  return (
    <div className="navbar">
      <ul>
        <li className="navbar-logo">🏠</li>
        <li>
          <p className="navbar-links">View Properties</p>
        </li>
        <li>
          <p className="navbar-links">Add a property</p>
        </li>
      </ul>
    </div>
  );
};

export default NavBar;
