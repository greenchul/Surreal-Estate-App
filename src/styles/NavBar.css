.navbar {
  background-color: black;
  color: white;
  height: 8vh;
}

a {
  color: white;
  text-decoration: none;
}

a:visited {
  color: white;
}

ul {
  display: flex;
  height: 100%;
  align-items: center;
}

li {
  list-style-type: none;
  display: inline;
  padding-right: 10px;
}
