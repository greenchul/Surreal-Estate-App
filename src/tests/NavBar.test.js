import { render, screen } from "@testing-library/react";
import React from "react";
import NavBar from "../components/NavBar";

describe("NavBar", () => {
  test("NavBar renders correctly", () => {
    const { asFragment } = render(<NavBar />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("NavBar links render correctly", () => {
    render(<NavBar />);
    const linkElement1 = screen.getByText("View Properties");
    const linkElement2 = screen.getByText("Add a property");

    expect(linkElement1).toBeInTheDocument();
    expect(linkElement2).toBeInTheDocument();
  });
});
