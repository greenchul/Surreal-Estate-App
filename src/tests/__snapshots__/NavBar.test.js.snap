// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavBar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <ul>
      <li
        class="navbar-logo"
      >
        üè†
      </li>
      <li>
        <p
          class="navbar-links"
        >
          View Properties
        </p>
      </li>
      <li>
        <p
          class="navbar-links"
        >
          Add a property
        </p>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
